### PR TITLE
v17: Migrate SubDAOs from distribution module -> gov

### LINKED ISSUE #829: Migrate SubDAOs to new admins
migrate subdaos to the actual gov account, not distr

- Core-1
- HackJuno
- JGF
- Delegations
- Communications

### PR DESCRIPTION
**Pull request by Reecepbcups**
Closes #829

Since this only runs on mainnet, there would be no way to test on Uni.

Future test case for LocalIC

---

Local Test Upgrade (example contract proof)
image

Logic used
```sh
# start v16 chain

  FILE_LOC=/home/reece/Desktop/blockchain_scripts/contracts/cw_template.wasm
  $OLDBINARY tx wasm store $FILE_LOC --from $KEY --gas 1000000 --chain-id $CHAINID --yes --keyring-backend $KEYRING --node $NODE

  CODE_ID=1
  # set to distr module account like mainnet
  txhash=`$OLDBINARY tx wasm instantiate $CODE_ID '{"count":0}' --from $KEY --gas 1000000 --chain-id $CHAINID --yes --keyring-backend $KEYRING --node $NODE --label="cw_template" --admin=juno1jv65s3grqf6v6jl3dp4t6c9t9rk99cd83d88wr --output=json | jq -r .txhash`
  contract_address=`$OLDBINARY q tx $txhash --node $NODE --output=json | jq -r .logs[0].events[1].attributes[0].value` && echo $contract_address
  
  # upgrade
    $NEWBINARY tx gov submit-legacy-proposal software-upgrade $UPGRADE_NAME --title "upgrading" --description "test upgrade" --deposit 1000000ujuno --upgrade-height $UPGRADE_HEIGHT --from $KEY --keyring-backend $KEYRING --chain-id $CHAINID --yes --broadcast-mode sync --node $NODE --no-validate --upgrade-info='{"binaries":{"linux/amd64":"https://linkhere.com/binary?checksum=sha256:aec070645fe53ee3b3763059376134f058cc337247c978add178b6ccdfb0019f"}}'
  $NEWBINARY tx gov vote "1" yes --from $KEY --keyring-backend=$KEYRING --node=$NODE --chain-id=$CHAINID --yes
  
  # start
  ```

### CODE REVIEW

**Comment by Reecepbcups** [comment]
cc @JakeHartnell lmk if there are any missing subdaos

File: app/upgrades/v17/upgrades.go [hunk 1]
```diff
@@ -60,6 +77,44 @@ func CreateV17UpgradeHandler(
...
 	}
 }
+
+func migrateChainOwnedSubDaos(ctx sdk.Context, logger log.Logger, ak authkeeper.AccountKeeper, ck *wasmkeeper.PermissionedKeeper) error {
+	logger.Info("migrating chain owned sub-daos")
+
+	distrAddr := ak.GetModuleAddress(distrtypes.ModuleName)
+
+	govAddr := ak.GetModuleAddress(govtypes.ModuleName)
+	govBech32, err := sdk.AccAddressFromBech32(govAddr.String())
```
Comment (Reecepbcups) [inline app/upgrades/v17/upgrades.go:98]: note to self this is wrong

**Comment by JakeHartnell** [comment]
> cc @JakeHartnell lmk if there are any missing subdaos

List looks good.

File: app/upgrades/v17/upgrades.go (hunk 1 above)
Comment (Reecepbcups) [reply app/upgrades/v17/upgrades.go]: fixed in 0badb5c26831517639ffe8bf09191c42c8f52d3c

File: app/upgrades/v17/upgrades.go [hunk 2]
```diff
@@ -60,6 +77,37 @@ func CreateV17UpgradeHandler(
 			return nil, err
 		}
 
+		// This function migrates all DAOs owned by the chain from the distribution module address -> the gov module.
+		// While they chains till owns it, technically it makes more sense to store them in the gov account.
```
Comment (dimiandre) [inline app/upgrades/v17/upgrades.go:81]: typo

**Review by dimiandre — APPROVED** [review]
looks good, addresses are correct

File: app/upgrades/v17/upgrades.go (hunk 2 above)
Comment (Reecepbcups) [reply app/upgrades/v17/upgrades.go]: 3c4ba996d84743bc6767adb252e0228e6ac280b2

**Pull request merged**